### PR TITLE
Improve type mapping documentation for Kafka connector

### PR DESCRIPTION
**Pull request by tlblessing**
## Description

Add a type mapping section. Relo SQL Support to bottom of topic for consistency.

### CODE REVIEW

File: docs/src/main/sphinx/connector/kafka.rst [hunk 1]
```diff
@@ -351,6 +351,8 @@ Field           Required  Type           Description
 ``message``     optional  JSON object    Field definitions for data columns mapped to the message itself.
 =============== ========= ============== =============================
 
+No other types are supported.
```
Comment (jhlodin) [inline docs/src/main/sphinx/connector/kafka.rst:354]: These are not type mappings; remove this line

File: docs/src/main/sphinx/connector/kafka.rst [hunk 2]
```diff
@@ -302,7 +302,7 @@ References a folder within Trino deployment that holds one or more JSON files
 This property is optional; the default is ``etc/kafka``.
 
 Table definition files
-""""""""""""""""""""""
+^^^^^^^^^^^^^^^^^^^^^^
```
Comment (jhlodin) [inline docs/src/main/sphinx/connector/kafka.rst:305]: Why was this changed?

File: docs/src/main/sphinx/connector/kafka.rst [hunk 3]
```diff
@@ -368,6 +370,8 @@ Field           Required  Type           Description
 ``fields``      required  JSON array     A list of field definitions. Each field definition creates a new column in the Trino table.
 =============== ========= ============== =============================
 
+No other types are supported.
```
Comment (jhlodin) [inline docs/src/main/sphinx/connector/kafka.rst:373]: These are not type mappings; remove this line

File: docs/src/main/sphinx/connector/kafka.rst [hunk 4]
```diff
@@ -397,6 +401,8 @@ Field           Required  Type      Description
 
 There is no limit on field descriptions for either key or message.
 
+No other types are supported.
```
Comment (jhlodin) [inline docs/src/main/sphinx/connector/kafka.rst:404]: These are not type mappings; remove this line

File: docs/src/main/sphinx/connector/kafka.rst [hunk 5]
```diff
@@ -700,6 +704,8 @@ The following Trino data types are supported by the CSV encoder:
 
 Column values are converted to strings before they are formatted as a CSV line.
 
+No other types are supported.
```
Comment (jhlodin) [inline docs/src/main/sphinx/connector/kafka.rst:707]: Move up to be immediately below the list

File: docs/src/main/sphinx/connector/kafka.rst [hunk 6]
```diff
@@ -817,6 +825,8 @@ The following table defines which temporal data types are supported by
 |                                     | ``seconds-since-epoch``                                                        |
 +-------------------------------------+--------------------------------------------------------------------------------+
 
+No other types are supported.
```
Comment (jhlodin) [inline docs/src/main/sphinx/connector/kafka.rst:828]: Not needed here, duplicate. Please remove

**Review by jhlodin — COMMENTED** [review]
Some changes needed. Please squash commits as well

**Comment by tlblessing** [comment]
Closing this PR since using the github gui borked it. Cole tried to help but it's beyond repair. Saved a backup and will do another PR.
